addons/stock_calendar/models/procurement.py

```diff
@@ -97,31 +91,29 @@ def _prepare_orderpoint_procurement(self, cr, uid, orderpoint, product_qty, date
             'group_id': group or orderpoint.group_id.id,
             }
 
-    def _get_previous_dates(self, cr, uid, orderpoint, start_date = False, context=None):
+    def _get_previous_dates(self, orderpoint, start_date=False):
         """
         Date should be given in utc
         """
-        calendar_obj = self.pool.get('resource.calendar')
-        att_obj = self.pool.get('resource.calendar.attendance')
-        context = context or {}
-        context['no_round_hours'] = True
+        ResourceCalendar = self.env['resource.calendar']
+        ResourceCalendarAttendance = self.env['resource.calendar.attendance']
         # Date should be converted to the correct timezone
-        start_date = self._convert_to_tz(cr, uid, start_date, context=context)
+        start_date = self.with_context(no_round_hours=True)._convert_to_tz(start_date)
         # First check if the orderpoint has a Calendar as it should be delivered at this calendar date
         purchase_date = False
         delivery_date = start_date
         if orderpoint.calendar_id and orderpoint.calendar_id.attendance_ids:
-            res = calendar_obj._schedule_days(cr, uid, orderpoint.calendar_id.id, -1, start_date, compute_leaves=True, context=context)
+            res = ResourceCalendar.with_context(no_round_hours=True)._schedule_days(orderpoint.calendar_id.id, -1, start_date, compute_leaves=True)
             if res and res[0][0] < start_date:
-                group_to_find = res[0][2] and att_obj.browse(cr, uid, res[0][2], context=context).group_id.id or False
+                group_to_find = ResourceCalendarAttendance.with_context(no_round_hours=True).browse(res[0][2]).group_id.id
                 delivery_date = res[0][0]
                 found_date = delivery_date
                 if orderpoint.purchase_calendar_id and orderpoint.purchase_calendar_id.attendance_ids:
                     while not purchase_date:
-                        found_date = found_date + relativedelta(days=-1) # won't allow to deliver within the day
-                        res = calendar_obj._schedule_days(cr, uid, orderpoint.purchase_calendar_id.id, -1, found_date, compute_leaves=True, context=context)
+                        found_date = found_date + relativedelta(days=-1)  # won't allow to deliver within the day
+                        res = ResourceCalendar.with_context(no_round_hours=True)._schedule_days(orderpoint.purchase_calendar_id.id, -1, found_date, compute_leaves=True)
                         for re in res:
-                            group = re[2] and att_obj.browse(cr, uid, re[2], context=context).group_id.id or False
+                            group = ResourceCalendarAttendance.with_context(no_round_hours=True).browse(re[2]).group_id.id
```

Comment:
no_roud_hours=True use 4 times so take one variable that is better solution like : 
```
ctx = dict(self.env.context or {})
ctx['no_round_hours'] = True
```